packages/main/src/components/FilterBar/FilterDialog.tsx

```diff
@@ -0,0 +1,252 @@
+        const filterBarItemRef = filterBarRefs.current[child.key];
+        let filterItemProps = {};
+        if (filterBarItemRef) {
+          filterItemProps = filterValue(filterBarItemRef, child);
+        }
+        return cloneElement(child as ReactElement<any>, {
+          children: {
+            ...child.props.children,
+            props: {
+              ...child.props.children.props,
+              ...filterItemProps,
+            },
+            ref: (node) => {
+              dialogRefs.current[child.key] = node;
+            },
+          },
+        });
+      });
+
+    return currentChildren;
+  }, [children, searchString, filterBarRefs]);
+
+  const handleCheckBoxChange = useCallback(
+    (element, toggledFilters) => (e) => {
+      if (handleSelectionChange) {
+        handleSelectionChange(enrichEventWithDetails(e, { element, checked: e.target.checked }));
+      }
+      setToggledFilters({ ...toggledFilters, [element.key]: e.target.checked });
+    },
+    [setToggledFilters, handleSelectionChange]
+  );
+  const renderGroups = useCallback(() => {
+    let groups = {};
+    Children.forEach(renderChildren(), (child) => {
+      const childGroups = child.props.groupName ?? 'default';
+      if (groups[childGroups]) {
+        groups[childGroups].push(child);
+      } else {
+        groups[childGroups] = [child];
+      }
+    });
+    return Object.keys(groups)
+      .sort((x, y) => (x === 'default' ? -1 : y === 'role' ? 1 : 0))
+      .map((item, index) => {
+        const filters = groups[item].map((el) => {
+          return (
+            <div className={classes.singleFilter} key={`${el.key}-container`}>
+              {el}
+              <CheckBox
+                checked={el.props.visibleInFilterBar || el.props.mandatory || el.type.displayName !== 'FilterGroupItem'}
+                onChange={handleCheckBoxChange(el, toggledFilters)}
+                disabled={el.props.mandatory || el.type.displayName !== 'FilterGroupItem'}
+              />
+            </div>
+          );
+        });
+        return (
+          <div className={classes.groupContainer} key={item}>
+            <FlexBox justifyContent={FlexBoxJustifyContent.SpaceBetween} alignItems={FlexBoxAlignItems.Center}>
+              <Title level={TitleLevel.H5} className={index === 0 ? classes.groupTitle : ''}>
+                {item === 'default' ? 'Basic' : item}
+              </Title>
+              {index === 0 && <Text wrapping={false}>Show on Filter Bar</Text>}
+            </FlexBox>
+            <div className={classes.filters}>{filters}</div>
+          </div>
+        );
+      });
+  }, [renderChildren, toggledFilters, handleCheckBoxChange]);
+
+  return (
+    <Dialog ref={dialogRef} onAfterClose={handleClose} header={renderHeader()} footer={renderFooter()}>
+      <div className={classes.dialog}>
+        {renderFBSearch && (
+          <div className={classes.fbSearch} ref={searchRef}>
+            <span />
```

Comment:
empty span?

Reply:
This is used as a placeholder to give the search row the same grid layout as the filter rows in the dialog box.